src/murmur3_32.rs

```diff
@@ -23,12 +25,12 @@ const N: u32 = 0xe654_6b64;
 /// use murmur3::murmur3_32;
 /// let hash_result = murmur3_32(&mut Cursor::new("hello world"), 0);
 /// ```
-pub fn murmur3_32<T: Read>(source: &mut T, seed: u32) -> Result<u32> {
+pub fn murmur3_32<T: Read>(mut source: &mut T, seed: u32) -> Result<u32> {
     let mut buffer: [u8; 4] = [0; 4];
     let mut processed = 0;
     let mut state = seed;
     loop {
-        match source.read(&mut buffer)? {
+        match read_bytes(&mut source, &mut buffer)? {
```

Comment:
You don't need to take a &mut of source.  It is already &mut T so you can pass it directly.  If you take this away then source in the args doesn't need to be mut anymore.  Same change is needed in all three files

Reply:
fixed